examples/gif_paintable/main.rs

```diff
@@ -0,0 +1,239 @@
+// A custom GDK paintable capable of rendering a GIF
+// The paintable makes uses of the awesome image
+// crate to read a gif file and transform it to a Vec<Frame>
+// which are then rendered by the paintable at different snapshots
+
+use gtk::{gdk, glib, graphene};
+use gtk::prelude::*;
+use gtk::subclass::prelude::*;
+use gtk::Application;
+use gtk::ApplicationWindow;
+use gtk::Picture;
+
+use image::{gif::GifDecoder, AnimationDecoder};
+
+use std::io::{Cursor, Read};
+use std::time::Duration;
+use std::ops::Deref;
+
+pub struct Frame {
+    pub width: i32,
+    pub height: i32,
+    pub texture: gdk::Paintable,
+    pub frame_duration: Duration,
+}
+
+impl Frame {
+    pub fn new(f: image::Frame) -> Self {
+        // transform the fractional time into a `Duration`
+        let (numerator, denominator) = f.delay().numer_denom_ms();
```

Comment:
There's a From implementation to convert from Delay into Duration that also handles fractions of milliseconds.

Some gifs also have 0ms delays set between frames and, by convention to match browsers, they should be treated as 100ms delays.

Reply:
Thats good advice. I knew there had to be some way to get a floating point representation but I didnt see a method for it. Also didn't know about the 0ms delays either.